ext/riverlea/schema/RiverleaStream.entityType.php

```diff
@@ -0,0 +1,137 @@
+<?php
+use CRM_riverlea_ExtensionUtil as E;
+
+return [
+  'name' => 'RiverleaStream',
+  'table' => 'civicrm_riverlea_stream',
+  'class' => 'CRM_riverlea_DAO_RiverleaStream',
+  'getInfo' => fn() => [
+    'title' => E::ts('RiverleaStream'),
+    'title_plural' => E::ts('RiverleaStreams'),
+    'description' => E::ts('Streams are configurable themes in the Riverlea Theme Framework'),
+    'log' => TRUE,
+  ],
+  'getIndices' => fn() => [
+    'index_stream_name' => [
+      'fields' => [
+        'name' => TRUE,
+      ],
+      'unique' => TRUE,
+      'add' => '6.1',
+    ],
+  ],
+  'getPaths' => fn() => [
+    // 'add' => 'civicrm/admin/riverlea/stream/create',
+    // 'update' => 'civicrm/admin/riverlea/stream/update#?RiverleaStream=[id]',
+    // 'delete' => 'civicrm/contact/view/delete?reset=1&delete=1&cid=[id]',
+  ],
+  'getFields' => fn() => [
+    'id' => [
+      'title' => E::ts('ID'),
+      'sql_type' => 'int unsigned',
+      'input_type' => 'Number',
+      'required' => TRUE,
+      'description' => E::ts('Unique RiverleaStream ID'),
+      'primary_key' => TRUE,
+      'auto_increment' => TRUE,
+    ],
+    'name' => [
+      'title' => ts('Machine-Name'),
+      'sql_type' => 'varchar(255)',
+      'input_type' => 'Text',
+      'description' => ts('Machine-name for this stream.'),
+    ],
+    'label' => [
+      'title' => ts('Label'),
+      'sql_type' => 'varchar(512)',
+      'input_type' => 'Text',
+      'required' => TRUE,
+      'description' => ts('User-facing name for this stream'),
+    ],
+    'description' => [
+      'title' => E::ts('Description'),
+      'sql_type' => 'text',
+      'input_type' => 'TextArea',
+      'description' => E::ts('Description of this stream'),
+      'default' => NULL,
+    ],
+    'is_reserved' => [
+      'title' => E::ts('Is Reserved?'),
+      'description' => E::ts('Reserved streams are not editable through the UI'),
+      'sql_type' => 'boolean',
+      'input_type' => 'CheckBox',
+      'required' => TRUE,
+      'default' => FALSE,
+    ],
+    'extension' => [
+      'title' => ts('Extension'),
+      'sql_type' => 'varchar(255)',
+      'input_type' => 'Select',
+      'description' => ts('Extension that provides this stream.'),
+      'default' => NULL,
+      // @todo why not work?
```

Comment:
Why not work?

Reply:
I think because the `civicrm_extension` table is slightly odd.

Reply:
I think it's kind of immaterial for now... for the time being this will always be provided by a `*.mgd.php` and it won't make sense to edit it.

Reply:
@ufundo which part doesn't work? The FK index in SQL? Or the join in Api4. If the former, it's now possible to declare an EntityRef without the FK index like so:
https://github.com/civicrm/civicrm-core/blob/d1a8138f23cfda44503d8ac8077a1e756e949a12/schema/Core/UserJob.entityType.php#L158-L164

Reply:
This still doesn't work in SearchKit / Formbuilder. I can't a) join Stream records to Extension records, b) in-place edit that field, c) add an afform filter.

I think maybe it's because `file` is not the primary key on `civicrm_extension` ? But we need to use a machine name (rather than `id`) because this is typically set in a managed record ( https://github.com/civicrm/civicrm-core/pull/32127/files#r2104616568 ).

I think for now it is fine to not have the entity_reference for now?